managed/services/management/node.go

```diff
@@ -136,10 +140,28 @@ func (s *NodeService) Register(ctx context.Context, req *managementpb.RegisterNo
 		return nil, e
 	}
 
-	apiKeyName := fmt.Sprintf("pmm-agent-%s-%d", req.NodeName, rand.Int63()) //nolint:gosec
-	_, res.Token, e = s.akp.CreateAdminAPIKey(ctx, apiKeyName)
-	if e != nil {
-		return nil, e
+	// get authorization from headers.
+	md, ok := metadata.FromIncomingContext(ctx)
+	if !ok {
+		msg := "Couldn't create Admin API Key: cannot get headers from metadata"
+		logrus.Errorln(msg)
+		res.Warning = msg
+		return res, nil
+	}
+	authorizationHeaders := md.Get("Authorization")
+	if len(authorizationHeaders) == 0 {
+		return nil, status.Error(codes.Unauthenticated, "Authorization error.")
+	}
+	headers := make(http.Header)
+	headers.Set("Authorization", authorizationHeaders[0])
+	if !s.akp.IsAPIKeyAuth(headers) {
+		apiKeyName := fmt.Sprintf("pmm-agent-%s-%d", req.NodeName, rand.Int63()) //nolint:gosec
+		_, res.Token, e = s.akp.CreateAdminAPIKey(ctx, apiKeyName)
+		if e != nil {
+			msg := fmt.Sprintf("Couldn't create Admin API Key: %s", e)
+			logrus.Errorln(msg)
```

Comment:
Don't we want to know which component we are logging from, i.e, `logrus.WithField("component", "node")`?

Reply:
not critical in this case, we don't have this error anywhere else

Reply:
anyway added